examples/weather/graphcast/compute_time_diff_std.py

```diff
@@ -0,0 +1,90 @@
+# SPDX-FileCopyrightText: Copyright (c) 2023 - 2024 NVIDIA CORPORATION & AFFILIATES.
+# SPDX-FileCopyrightText: All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import os
+from modulus.datapipes.climate import ERA5HDF5Datapipe
+from modulus.distributed import DistributedManager
+
+import hydra
+from hydra.utils import to_absolute_path
+from omegaconf import DictConfig
+import numpy as np
+
+from loss.utils import grid_cell_area
+
+
+@hydra.main(version_base="1.3", config_path="conf", config_name="config")
+def main(cfg: DictConfig) -> None:
+
+    # initialize distributed manager
+    DistributedManager.initialize()
+    dist = DistributedManager()
+
+    # instantiate the training datapipe
+    DataPipe = ERA5HDF5Datapipe  # [T,num_channel, 721, 1440], grid features
+    datapipe = DataPipe(
+        data_dir=to_absolute_path(os.path.join(cfg.dataset_path, "train")),
+        stats_dir=to_absolute_path(os.path.join(cfg.dataset_path, "stats")),
+        channels=[i for i in range(cfg.num_channels_climate)],
+        latlon_resolution=cfg.latlon_res,
+        num_samples_per_year=cfg.num_samples_per_year_train,
+        num_steps=1,
+        batch_size=1,
+        num_workers=cfg.num_workers,
+        device=dist.device,
+        process_rank=dist.rank,
+        world_size=dist.world_size,
+        shuffle=False,
+    )
+    print(f"Loaded training datapipe of length {len(datapipe)}")
+
+    area = (
+        grid_cell_area(torch.linspace(-90, 90, steps=cfg.latlon_res[0]), unit="deg")
```

Comment:
is this always correct or in-sync with the model definition? can we maybe add an utility that defines these lat-lon meshes? in the training, the "area" is computed from the `model.latitude` values directly

side-note: the area function expects the inputs to be in a -90/90 range anyways, a 0/180 range e.g. would require to replace the cos-term with a sin-term.

Reply:
Yes, this is in sync with the model definition. Although I think both are inconsistent with the ERA5 grid, which goes from 90 to -90. However, because cell area is proportional to cosine of latitudes, this does not change the final values.

Reply:
The main issue arises from this being inconsistent with the model, i.e. if you would have different area weights due to slightly different definitions.

Reply:
I guess for now, this should be fine though.